controller/lib/core/blower_fsm.cpp

```diff
@@ -143,14 +143,18 @@ BlowerSystemState BlowerFsm::DesiredState(Time now, const VentParams &params,
 
   std::visit([&](auto &fsm) { return fsm.Update(now, inputs); }, fsm_);
 
-  if ((params.mode == VentMode_OFF && !std::holds_alternative<OffFsm>(fsm_)) ||
+  if (params.mode == VentMode_OFF ||
       std::visit([&](auto &fsm) { return fsm.Finished(); }, fsm_)) {
-    // Set is_new_breath to true even when the ventilator transitions from on
-    // to off.  It's a little arbitrary, but for the most part, is_new_breath
-    // is used to mark breath boundaries rather than simply signal the
-    // beginning of a breath, and it would be weird if the last breath before
-    // turning off the ventilator appeared to continue indefinitely.
-    is_new_breath = true;
+    // Set is_new_breath to true only when entering a new cycle, not if this
+    // is the continuation of the Off Fsm.
+    // This means setting it to true on the transition from On to Off, which
+    // allows the controller to keep track of volume once the ventilator is
+    // turned Off, assuming the user will stop the ventilator when volume
+    // is close to 0.
```

Comment:
The meaning of "entering a new cycle" is not obvious to me.  I think you mean "starting a new breath".  But then that's also confusing comment, clearly is_new_breath would be set to true when starting a new breath...

I think we can simplify this code some.  The OffFsm always returns `Finished() == true`, so we don't need the `params.mode == VentMode_OFF` check any longer.  Then maybe we change the code to

```
// For the purposes of is_new_breath, treat multiple occurrences of OffFsm as a "single breath".
is_new_breath = !std::holds_alternative<OffFsm>();
```

Reply:
I like it a lot your way! Will do.

Reply:
OK, turns out you need the params.mode == VentMode_OFF to take the Off command immediately... will still use the rest of your suggestion, which makes everything much easier to follow.